torch/csrc/jit/script/module.h

```diff
@@ -86,6 +86,28 @@ struct TORCH_API Method {
     return stack.front();
   }
 
+  void saveInstructions(
```

Comment:
This is tightly coupling internal implementation details of our interpreter with an ad-hoc format for serialization. Furthermore, there are no tests it even works.

This cannot be merged as is. It will make development of the interpreter difficult (did I break an ad hoc serialization format I am not familiar with?).

Reply:
The test to cover this function is  in src/pytorch/test/cpp/jit/test_lite_executor.h in this pull. Since it's the first diff of a WIP stack, currently the test just checks if there is something in the output stream. After we added the parser and executor, we could compare the results of a saved/loaded module to the original module (steps included in the "TODO" list in test_lite_executor.h. First time to use git pull and I will put more explanations. Thanks for the suggestion! 
I'm trying to make separate functions for instruction-related only. This function is an entry point to dump the low level items (for example, execution_plan and the instruction list). As we discussed, the instruction part is low-level and unlikely couple with the interpreter work that is under development. Could you point me some design docs/plans on "an ad-hoc format for serialization", and elaborate how does this part couple with it? Any further suggestions on how to decouple while accessing to the instructions that we want to serialize?